src/js/chrome/create-chrome.js

```diff
@@ -0,0 +1,81 @@
+import { loadNav } from '../nav/globalNav';
+import qe from './iqeEnablement';
+import sourceOfTruth from '../nav/sourceOfTruth';
+import { spinUpStore } from '../redux-config';
+import { chromeInit, bootstrap } from './entry';
+import initializeJWT from './initialize-jwt';
+import { createFetchPermissionsWatcher } from '../rbac/fetchPermissions';
+
+/**
+ * Create a chrome API instance
+ * @param {object} jwt JWT auth functions
+ * @param {object} insights existing insights instance
+ */
+const createChromeInstance = (jwt, insights) => {
+    const { actions: { chromeNavUpdate } } = spinUpStore();
+    const libjwt = jwt;
+    const chromeInstance = {
+        cache: undefined
+    };
+
+    const jwtResolver = initializeJWT(libjwt, chromeInstance);
```

Comment:
Might be a better to use then instead of mutating `chromeInstance.cache` when init is done
```JS
let chromeCache;
const jwtResolver = initializeJWT(libjwt);
jwtResolver.finally(() => {
  chromeCache = new CacheAdapter(
    'chrome-store',
    `${decodeToken(libjwt.jwt.getEncodedToken())?.session_state}-chrome-store`
  );
})
```

Reply:
Or going full promise
```JS
const chromeCache = jwtResolver.finally(() => new CacheAdapter(
    'chrome-store',
    `${decodeToken(libjwt.jwt.getEncodedToken())?.session_state}-chrome-store`
  )
);
```

And whenever `chromeCache` is called wrap it with either `then` or `await`

Reply:
This won't work the chromeCache will not mutate in functions where the cache is passed as a parameter. It will create a copy instead of keeping the pointer.

Reply:
Ah, yeah. The problem with pointers and values of JS :disappointed: I don't think promisifying it won't help us with readability either... Keep it as is